mission_control/navigator_object_recognition/object_classification.py

```diff
@@ -0,0 +1,189 @@
+        min_dist = 10000
+        for b in self.all_objects:
+            obj_pose = nt.rosmsg_to_numpy(b.position)
+            dist = np.linalg.norm(obj_pose - point)
+            print dist
+            if dist < min_dist:
+                min_obj = b
+                min_dist = dist
+
+        marker = Marker()
+        marker.header.stamp = nh.get_time()
+        marker.header.seq = 1
+        marker.header.frame_id = "enu"
+        marker.id = min_obj.id
+        marker.pose.position = min_obj.position
+        marker.type = marker.SPHERE
+        marker.action = marker.ADD
+        marker.scale.x = 1.0
+        marker.scale.y = 1.0
+        marker.scale.z = 1.0
+        marker.color.r = 1.0
+        marker.color.g = 0.0
+        marker.color.b = 0.0
+        marker.color.a = 1.0
+        self.pub_object_searching.publish(marker)
+        yield nh.sleep(1)
+        x = yield util.nonblocking_raw_input("What object is this? " + str(marker.id))
+
+        if(x == 'skip'):
+            self.currently_classifying = False
+            return
+        self.classified_ids[min_obj.id] = x
+
+        obj = PerceptionObject()
+        obj.name = self.classified_ids[min_obj.id]
+        obj.position = min_obj.position
+        obj.size.x = min_obj.height
+        obj.size.y = min_obj.width
+        obj.size.z = min_obj.depth
+        obj.id = min_obj.id
+        self.pub_obj_found.publish(obj)
+
+        marker_del = Marker()
+        marker_del.action = 3  # This is DELETEALL
+        self.pub_object_searching.publish(marker_del)
+
+        self.currently_classifying = False
+
+    def new_objects(self, buoy_array):
+        if self.human_mode and not self.currently_classifying:
+            self.all_objects = buoy_array.buoys
+            return
+
+        if self.currently_classifying:
+            return
+
+        if self.position is None:
+            return
+
+        self.currently_classifying = True
+        buoys = buoy_array.objects
+
+        for b in buoys:
+            v = b.height * b.width * b.depth
+            if b.id not in self.classified_ids.keys() or 'un' in self.classified_ids[b.id]:
+                dist = np.linalg.norm(nt.rosmsg_to_numpy(b.position) - self.position)
+                v = b.height * b.width * b.depth
+                h = b.height
+                dir_vec = (nt.rosmsg_to_numpy(b.position) - self.position) / dist
+                dir_vec[2] = 0
+                yaw = tf.transformations.euler_from_quaternion(self.rot)[2]
+                nav_vec = [np.cos(yaw), np.sin(yaw), 0]
+                mydot = np.dot(dir_vec, nav_vec)
+                angle = np.arccos(mydot)
+
+                print "id: ", b.id, yaw, mydot, angle
+
+                if angle > (3.14 / 6):
+                    continue
+
+                print "id _ made it: ", b.id
+                print "----------"
+
+                if dist > 25:
+                    x = 'un' + str(b.id)
+
+                elif v > 1 and v < 3 and h < 2:
```

Comment:
Good idea, however, I am just going to remove this code and make this a a manual object classification tool only. Since Ira is doing the automatic one.